### PR TITLE
doc improvement: move wifi fw page

### PR DESCRIPTION
**Pull request by kittydepa**
Follow-up to https://github.com/nrfconnect/sdk-nrf/pull/14075

- Move Wi-FI FW patch update page from DFU page to Developing with nRF70 Series
- Added a release note entry for what was covered in https://github.com/nrfconnect/sdk-nrf/pull/14075
- Added cross-ref links to 2 device specific guides in the DFU page

### CODE REVIEW

**Review by ioannisg — APPROVED** [review]
lgtm (high-level review only)

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -658,7 +665,11 @@ Matter samples
 
   * :ref:`ug_matter_configuring_read_client` in most Matter samples using the new :kconfig:option:`CONFIG_CHIP_ENABLE_READ_CLIENT` Kconfig option.
   * WPA Supplicant advanced features in all Matter samples using the :kconfig:option:`WPA_SUPP_ADVANCED_FEATURES` Kconfig option.
-    This saves roughly 25 KB of FLASH memory for firmware images with Wi-Fi support.
+    This saves roughly 25 kB of FLASH memory for firmware images with Wi-Fi support.
+
+* :ref:`matter_lock_sample` sample:
+
+  * Added support for Wi-Fi firmware upgrade on external memory.
```
Comment (ArekBalysNordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:672]: This is valid only for `nrf5340dk_nrf5340 + nrf7002EK`, so could you mention it somewhere?

The second question, could you please add a similar note to `Matter Bridge` Application? Regarding that application, I've added support for `nrf5340dk_nrf5340 + nrf7002EK` (Previously we had support for `nrf7002dk_nrf5340` only), and now we have the same support for Wi-Fi firmware upgrade on external memory there (also valid for  `nrf5340dk_nrf5340 + nrf7002EK` only )

Comment (kittydepa) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Added, please confirm if it is correctly described.

**Review by annwoj — APPROVED** [review]

File: doc/nrf/config_and_build/dfu/index.rst
Comment (nvlsianpu) [inline doc/nrf/config_and_build/dfu/index.rst:1]: What about adding to this page subsection with will only point platforms specific features of DFU solutions (Platform specific features/support?). I see two features which can be listed there: the wifi patch (this), the nRF53 external QSPI xip.

Comment (kittydepa) [reply doc/nrf/config_and_build/dfu/index.rst:1]: Discussing with Anna - will get back to this before doc improvement deadline (7 March)

Comment (nvlsianpu) [reply doc/nrf/config_and_build/dfu/index.rst:1]: OK

**Review by nvlsianpu — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -397,6 +404,7 @@ Matter Bridge
   * Support for Bluetooth LE Security Manager Protocol that allows to establish secure session with bridged Bluetooth LE devices.
   * Callback to indicate the current state of Bluetooth LE Connectivity Manager.
     The current state is shown by LED 2.
+  * Support for Wi-Fi firmware upgrades on external memory, for the nRF5340 DK with the nRF7002 EK attached.
```
Comment (ArekBalysNordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:407]:
```suggestion
    * Support for the nRF5340 DK with the nRF7002 EK attached.
    * Support for Wi-Fi firmware upgrades on external memory, for the nRF5340 DK with the nRF7002 EK attached.
```

**Review by ArekBalysNordic — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -216,6 +216,13 @@ nRF IEEE 802.15.4 radio driver
 Wi-Fi
 -----
 
+* Added:
+
+  * The :kconfig:option:`NRF_WIFI_FW_PATCH_DFU` Kconfig option to enable DFU support for nRF70 Series devices.
+    This allows firmware patches for signed images to be sent to multi-image targets over Wi-Fi.
+
+  * The :ref:`ug_nrf70_fw_patch_update` user guide in the :ref:`ug_nrf70_developing` section that describes how to perform a DFU for firmware patches using Wi-Fi.
```
Comment (richabp) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:224]: I would suggest to move line 224 under Documentation section after line 1367, as this doc is not added under Protocols section.

**Review by richabp — CHANGES_REQUESTED** [review]

File: doc/nrf/config_and_build/dfu/index.rst
Comment (kittydepa) [reply doc/nrf/config_and_build/dfu/index.rst:1]: Will most likely add cross-referencing at the very least - thank you for pointing this out.

**Review by richabp — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

File: doc/nrf/config_and_build/dfu/index.rst
Comment (kittydepa) [inline doc/nrf/config_and_build/dfu/index.rst:1]: @annwoj Please take a look when you get the chance

File: doc/nrf/config_and_build/dfu/index.rst [hunk 4]
```diff
@@ -73,11 +73,15 @@ Based on these criteria, the |NCS| offers support for the following DFU alternat
       - Library in the |NCS| that provides functions for downloading a firmware file as an upgrade candidate to the DFU target. The library is often used by IoT libraries, such as the :ref:`lib_nrf_cloud` library.
       - OTA (LTE, Wi-Fi)
 
-See the following user guides for an overview of topics related to firmware updates with the |NCS|:
+See the following pages for device-specific guides related to DFU:
+
+* :ref:`qspi_xip` - for the nRF5340 DK
```
Comment (annwoj) [inline doc/nrf/config_and_build/dfu/index.rst:78]: TBC: shouldn't this say SoC?

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -218,6 +218,9 @@ nRF IEEE 802.15.4 radio driver
 Wi-Fi
 -----
 
+* Added the :kconfig:option:`NRF_WIFI_FW_PATCH_DFU` Kconfig option to enable DFU support for nRF70 Series devices.
+    This allows firmware patches for signed images to be sent to multi-image targets over Wi-Fi.
```
Comment (annwoj) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:222]:
```suggestion
* Added the :kconfig:option:`NRF_WIFI_FW_PATCH_DFU` Kconfig option that enables DFU support for nRF70 Series devices.
  This allows firmware patches for signed images to be sent to multi-image targets over Wi-Fi.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 6]
```diff
@@ -698,8 +703,12 @@ Matter samples
 * Disabled:
 
   * :ref:`ug_matter_configuring_read_client` in most Matter samples using the new :kconfig:option:`CONFIG_CHIP_ENABLE_READ_CLIENT` Kconfig option.
-  * WPA supplicant advanced features in all Matter samples using the :kconfig:option:`CONFIG_WPA_SUPP_ADVANCED_FEATURES` Kconfig option.
-    This saves roughly 25 KB of FLASH memory for firmware images with Wi-Fi support.
+  * WPA supplicant advanced features in all Matter samples using the :kconfig:option:`WPA_SUPP_ADVANCED_FEATURES` Kconfig option.
+    This saves roughly 25 kB of FLASH memory for firmware images with Wi-Fi support.
```
Comment (annwoj) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:707]:
```suggestion
    This saves roughly 25 KB of flash memory for firmware images with Wi-Fi support.
```
@kittydepa tbc - don't think I saw this capitalized anywhere else

File: doc/nrf/config_and_build/dfu/index.rst (hunk 4 above)
Comment (nvlsianpu) [reply doc/nrf/config_and_build/dfu/index.rst:78]: SoC

**Pull request merged**